astropy/io/fits/src/compressionmodule.c

```diff
@@ -233,159 +235,132 @@ int compress_type_from_string(char* zcmptype) {
 }
 
 
+PyObject *
+get_header_value(PyObject* header, char* key) {
+    PyObject* hdrkey;
+    PyObject* hdrval;
+    hdrkey = PyUnicode_FromString(key);
+    if (hdrkey == NULL) {
+        return NULL;
+    }
+    hdrval = PyObject_GetItem(header, hdrkey);
+    Py_DECREF(hdrkey);
+    PyErr_Clear();
+    return hdrval;
+}
+
+
 // TODO: It might be possible to simplify these further by making the
 // conversion function (eg. PyString_AsString) an argument to a macro or
 // something, but I'm not sure yet how easy it is to generalize the error
 // handling
+/* The get_header_* functions resemble "Header.get" where "def" is the default
+   value, "keyword" is a string representing the header-key and the result is
+   stored in "val".
+   The function returns 0 on success, 1 if the header didn't have the keyword
+   and the default was applied and -1 (with an exception set) if an Exception
+   happened (like a MemoryError or Overflow).
+*/
 int get_header_string(PyObject* header, char* keyword, char* val, char* def) {
-    PyObject* keystr;
-    PyObject* keyval;
-    PyObject* tmp;  // Temp pointer to decoded bytes object
-    int retval;
-
-    keystr = PyUnicode_FromString(keyword);
-    keyval = PyObject_GetItem(header, keystr);
-
-    if (keyval != NULL) {
-        // FITS header values should always be ASCII, but Latin1 is on the
-        // safe side
-        tmp = PyUnicode_AsLatin1String(keyval);
-        strncpy(val, PyBytes_AsString(tmp), 72);
-        Py_DECREF(tmp);
-        retval = 0;
-    }
-    else {
-        PyErr_Clear();
+    PyObject* keyval = get_header_value(header, keyword);
+
+    if (keyval == NULL) {
         strncpy(val, def, 72);
```

Comment:
It might be nice to "un-magic" the number `72` (here and below) with a `#define`.

Reply:
Do you have a good suggestion for a name? 

It includes the python, numpy and cfitsio namespaces, so we probably should use a prefix to avoid accidental name clashes.

Reply:
It's good to choose a unique name but C doesn't have name spaces in the same way that Python does (unfortunately). It looks like the intent is maybe for something like `#define MAX_HEADER_VAL_LEN`. However, this should also be used to define the length of the destination buffers in `configure_compression` and elsewhere, so maybe this belongs in a separate PR. Also, in looking at the way this is called in `configure_compression` and elsewhere, I'm not entirely sure whether the buffers are being null terminated properly in all cases, or whether that's necessary here. But again, that's not necessarily relevant to your changes.

Reply:
okay, let's leave that for another PR. Could you open an issue about it so we don't forget?

Reply:
I can do that. Do you think it should be a PR for this particular change, or should we also hunt for other magic values in FITS C extension code as a code clean-up project?

Reply:
Not sure if we should clean them up, but we should probably check if it's properly null-terminated. :)